src/librustc_mir/util/liveness.rs

```diff
@@ -281,24 +283,25 @@ pub fn categorize<'tcx>(context: PlaceContext<'tcx>, mode: LivenessMode) -> Opti
     }
 }
 
-struct DefsUsesVisitor {
+struct DefsUsesVisitor<V: Idx> {
     mode: LivenessMode,
-    defs_uses: DefsUses,
+    defs_uses: DefsUses<V>,
 }
 
 #[derive(Eq, PartialEq, Clone)]
-struct DefsUses {
-    defs: LocalSet,
-    uses: LocalSet,
+struct DefsUses<V: Idx>
+{
+    defs: LocalSet<V>,
+    uses: LocalSet<V>,
 }
 
-impl DefsUses {
+impl<V: Idx> DefsUses<V> {
     fn clear(&mut self) {
         self.uses.clear();
         self.defs.clear();
     }
 
-    fn apply(&self, bits: &mut LocalSet) -> bool {
+    fn apply(&self, bits: &mut LocalSet<V>) -> bool {
         bits.subtract(&self.defs) | bits.union(&self.uses)
     }
 
```

Comment:
In the code (not shown as part of the diff) below, like this:

```rust
if let Some(v_index) = self.map.from_local(index) {
    self.uses.remove(&index);
    self.defs.add(&index);
}
```
```

I presume we are now getting type errors. This is because `uses` and `defs` are sets of `V` indices, but we are using a `Local` -- wrong type. This is where that new trait comes in. We would convert like so:

```rust
```